### PR TITLE
Further expand `TRANSLATING.md` with more tips and guidance

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- New or updated content

#### Description

This PR adds a few more things to our i18n guide:

- Add guidance for glossaries/style guides at `i18n/README.md` 

- Add code samples guidance.

- Emphasize Discord participation.

- Emphasize reviews.

- Mention that now that the "Translate this page" button disappears for pages that are not available, you can use the Docs as a way to quickly see if a page is available.

- Reorder headings.

- Review tips and shoutout!

### CODE REVIEW

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:7]:
```suggestion
To get involved in our translation efforts, we **highly recommend** joining our Discord chat](https://astro.build/chat) first. This way, we can get you up to speed with our process and give you the opportunity to chat with your language's translation team.
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:9]:
```suggestion
Most of our internationalization decisions and discussions happen on Discord. So, it's the best way to find out which patterns and recommendations your language's translation team follows before making your own PRs. You can even become a part of the decision-making process.
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:21]:
```suggestion
See our automated [Translation Status Overview issue](https://github.com/withastro/docs/issues/438) for a quick list of which pages are missing a translation or need updating to match a change to the English version.
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:24]:
```suggestion
> Please do not translate any pages without first checking their status in our [Overview Issue](https://github.com/withastro/docs/issues/438)! If a page is not listed here as needing a translation or an update, we can not accept your PR.
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:26]:
```suggestion
Not every page is marked as "ready to translate." So, even if you find a page that is not yet translated on the Docs site, you must first confirm that it is on the list of available pages to translate. Do not translate documents that are missing:

- A "Translate this page" button in the Docs site.
- The `i18nReady: true` frontmatter property in its Markdown file.
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:145]:
```suggestion
Translators are free to create and mantain a glossary, style guide and other tips for their language's translation squad. This is a great way to keep translations consistent across contributors and to centralize team decisions. You can find it (or create it) inside the language's `i18n` folder as a `README.md` file.
```

**Review by sarah11918 — COMMENTED** [review]
Hey @Yan-Thomas! These are great! I didn't describe each of my suggestions below; and I'm not necessarily committed to them. They are all just kind of ideas at the point for you to play around with!  💜

**Comment by Yan-Thomas** [comment]
Thanks for all the input @sarah11918 💜! I restructured the headings a bit and added review tips. I think I'm mostly satisfied with the changes we did in this PR, so I'm opening it for review and will see if we may want to do more changes after.

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:7]: Broken link tag!

```suggestion
To get involved in our translation efforts, we **highly recommend** [joining our Discord chat](https://astro.build/chat) first. This way, we can get you up to speed with our process and give you the opportunity to chat with your language's translation team.
```

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:9]: “so” is more of a conjunction, so we could either elide these first two sentences, or I think just break them up a little differently. How does this feel?

```suggestion
Most of our internationalization decisions and discussions happen on Discord. Joining us there is the best way to find out which patterns and recommendations your language's translation team follows before making your own PRs. You can even become a part of the decision-making process.
```

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:19]: In general, we try to avoid judgements like “simply” as they might imply something should be simple to do, which can be frustrating if someone is not finding it simple! I’d suggest removing that in this case.

```suggestion
To find translations pending review, you can filter through this repo's [Pull Requests with the i18n label](https://github.com/withastro/docs/pulls?q=is%3Apr+is%3Aopen+label%3Ai18n).
```

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:97]: I wonder if we want to move all the Translation Tips _above_ “Adding a new language”? That section is really only aimed at maintainers, so might be better to relegate it to the bottom of the document.

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:195]: I think “aware” might be better here; “wary” implies more “be fearful” (which depends on how you feel about Hippo’s code highlighting syntax 😆)

```suggestion
Be aware that if code samples are being translated, you may need to update some of the code sample's highlighted lines. Read the [Code Samples section](https://github.com/withastro/docs/blob/main/WRITING.md#code-samples) in our Writing Guide to know more about our syntax.
```

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:265]: Thinking about priorities again, maybe we want to move this even _above_ Translation Tips?

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:269]: Do you think it would be helpful to explain what LGTM means? (I half thought about mentioning our other fun versions of LGTM, but maybe in this guide it’s best to be clear and simple.)

```suggestion
**When you think a PR is good to be merged**, approve the PR through GitHub's "Review Changes" button or leave a "**LGTM!**" in the comments. (“LGTM” is an abbreviation of “Looks Good to Me” often used to approve pull requests.)
```

File: TRANSLATING.md
Comment (delucis) [inline TRANSLATING.md:284]: Love this list. Super helpful!

**Review by delucis — COMMENTED** [review]
Really helpful work @Yan-Thomas!

File: TRANSLATING.md
Comment (Yan-Thomas) [reply TRANSLATING.md:19]: oh, good point!

File: TRANSLATING.md
Comment (Yan-Thomas) [reply TRANSLATING.md:269]: I don't see why not. Sure could help someone that isn't familiarized with the abbreviation, good idea!

File: TRANSLATING.md
Comment (Yan-Thomas) [reply TRANSLATING.md:97]: Yeah, "Adding a new language" is more of a one-time thing, but Review/Translation Tips can be revisited many times by contributors, so I agree with moving these up!

File: TRANSLATING.md
Comment (Yan-Thomas) [reply TRANSLATING.md]: Moved!

**Comment by Yan-Thomas** [comment]
Thanks for all the great ideas and suggestions @delucis! 💜🙌

**Review by delucis — APPROVED** [review]
LGTM! Thanks @Yan-Thomas

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:107]:
```suggestion
We love our reviewers! Reviewing PRs is an important task — it's thanks to the efforts of our reviewers that we can guarantee consistent, high-quality translations. Many projects track PRs that you submit, but we also celebrate review stats, visible on your very own [Astro Badge](https://astro.badg.es)! Visit our Discord and you'll see that we shout out every PR merged with a list contributors who helped with review comments.

We even have an entire section in our Maintainer's Guide about [how to manually add reviewer's names to commit messages](https://github.com/withastro/docs/blob/main/MAINTAINERS.md#getting-co-authored-by-commit-message-name-and-email) before merging in case they are not automatically included!

So, if you're interested in helping review translation PRs, thank you! We really appreciate the effort, and we put an effort into showing it!
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:109]: I think I would put this after the checklist, since approving it happens AFTER they've gone through the checklist!

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:116]:
```suggestion
It can be confusing to know what exactly should be reviewed! Mostly, what we are looking for is another pair of eyes to catch any obvious mistakes. If that is all you do, you have been a HUGE help!

Just ask yourself: does anything seem "out of place" or "unusual" when I read it? Are there typos or any unusual words choices that I would not expect to read? This might not mean the translation is "wrong" but *is* worth mentioning if a word choice is distracting when reading documentation.

If you want to take your reviews to the next level, here are some more questions you can ask yourself while reviewing translations:
```

File: TRANSLATING.md
Comment (sarah11918) [inline TRANSLATING.md:178]:
```suggestion
We have lots of code samples throughout our docs, and although we **recommend translating comments**, as they give a contextual clue of what's happening in the code, each language is **free to decide** whether or not they want to translate titles, variables, string values, function names, etc.

Be aware that if code samples are being translated, you may need to update some of the code sample's highlighted lines. Read the [Code Samples section](https://github.com/withastro/docs/blob/main/WRITING.md#code-samples) in our Writing Guide to know more about our syntax.
```

**Review by sarah11918 — COMMENTED** [review]
Amazing addition of tips, Yan!!  Some suggestions for you to consider below! 💜

File: TRANSLATING.md
Comment (Yan-Thomas) [reply TRANSLATING.md]: Fair point! Just applied this change

**Comment by sarah11918** [comment]
LGTM!

**Pull request merged**
